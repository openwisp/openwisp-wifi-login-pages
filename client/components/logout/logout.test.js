/* eslint-disable prefer-promise-reject-errors */
import {shallow} from "enzyme";
import PropTypes from "prop-types";
import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import getConfig from "../../utils/get-config";
import logError from "../../utils/log-error";
import Logout from "./logout";

jest.mock("axios");
jest.mock("../../utils/get-config");
jest.mock("../../utils/log-error");
logError.mockImplementation(jest.fn());

const defaultConfig = getConfig("default");

const createTestProps = (props) => {
  return {
    language: "en",
    orgSlug: "default",
    logoutPage: defaultConfig.components.logout,
    authenticate: jest.fn(),
    ...props,
  };
};

describe("<Logout /> rendering", () => {
  let props;

  it("should render correctly", () => {
    props = createTestProps();
    const renderer = new ShallowRenderer();
    const component = renderer.render(<Logout {...props} />);
    expect(component).toMatchSnapshot();
  });
});

describe("<Logout /> interactions", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    Logout.contextTypes = {
      setLoading: PropTypes.func,
      getLoading: PropTypes.func,
    };
  });

  it("should set user authenticated when log in again is clicked", () => {
    props = createTestProps();
    wrapper = shallow(<Logout {...props} />, {
      context: {setLoading: jest.fn()},
    });
    const loginUser = jest.spyOn(wrapper.instance(), "loginUser");
    wrapper.find(".button").simulate("click", {});
    expect(loginUser).toHaveBeenCalled();
  });
});
