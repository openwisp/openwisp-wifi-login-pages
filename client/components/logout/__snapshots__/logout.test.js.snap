// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> rendering should render correctly 1`] = `
<React.Fragment>
  <div
    className="container content"
    id="logout"
  >
    <div
      className="inner"
    >
      <div
        className="main-column w-100"
      >
        <div
          className="inner"
        >
          <h2>
            Logout successful
          </h2>
          <div
            className="links row"
          >
            <Link
              className="button partial"
              onClick={[Function]}
              to="/default/status"
            >
              Login again
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</React.Fragment>
`;

exports[`<Logout /> rendering with placeholder translation tags should render translation placeholder correctly 1`] = `
<React.Fragment>
  <div
    className="container content"
    id="logout"
  >
    <div
      className="inner"
    >
      <div
        className="main-column w-100"
      >
        <div
          className="inner"
        >
          <h2>
            LOGOUT_CONTENT
          </h2>
          <div
            className="links row"
          >
            <Link
              className="button partial"
              onClick={[Function]}
              to="/default/status"
            >
              LOGIN_AGAIN
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</React.Fragment>
`;
